aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -844,6 +844,12 @@ Tensor scalar_tensor(const Scalar& s,
     std::optional<Layout> layout,
     std::optional<Device> device,
     std::optional<bool> pin_memory) {
+  // NB: It's always wrong to try to create a scalar tensor with the jagged layout.
+  // Rather than fix this everywhere, just use the strided layout and let NJT handle
+  // scalar tensor broadcasting.
+  if (layout.has_value() && layout == at::kJagged) {
```

Comment:
Technically, you don't need to do a has_value check, since std::nullopt can never `==` at::kJagged
So:
```
  if (layout == at::kJagged) {
```
would be equivalent.